src/java.base/share/classes/javax/crypto/KDF.java

```diff
@@ -479,6 +477,24 @@ private static KDF handleException(NoSuchAlgorithmException e)
         throw e;
     }
 
+    // Rethrows the IAPE thrown by an implementation, adding an explanation
+    // on in which situation it fails.
+    private void rethrow(InvalidAlgorithmParameterException e)
+            throws InvalidAlgorithmParameterException {
+        var source = serviceIterator == null
+                ? "specified" : "previously selected";
+        if (!skipDebug && pdebug != null) {
+            pdebug.println("A " + this.getAlgorithm()
+                    + " derivation cannot be performed "
+                    + "using the supplied derivation "
+                    + "inputs, using the " + source + " "
+                    + theOne.provider().getName()
```

Comment:
Suggest rewording as: "inputs with the " + source + " " + theOne.provider().getName() + " provider."

Reply:
The exception message following uses `"The " + source + " provider " + theOne.provider.getName()`. Shall I also switch to `"The " + source + " " + theOne.provider.getName() + " provider "`?